[LCS-334] - Update Urls & relevant associated field names

sulthan-ahmed opened the pull request:
- current-property-address to tenant-current-address
- landlord-agent to landlord-or-agent

lennym commented:
The ticket says to change the urls, but you've changed all of the field names as well? Is there a reason for that?

sulthan-ahmed commented:
@lennym maybe my eyes are deceiving me but where have I tried the field names? 

I did change the address key which will affect the field names in the hof-behaviour-address-lookup but I thought I should change that because 1. I thought it needs to match with the url to get it to work as part of the config.  2. Keep it consistent.

lennym commented:
Your eyes are deceiving you then. Most of the changes to acceptance tests are changes to the names/ids of the fields being filled in. 

If you *haven't* changed the field names then these changes shouldn't have been made, as you're now instructing the tests to fill out fields that don't exist.

lennym commented on apps/right-to-rent-check/acceptance/features/confirm/agent-not-in-property.js:
```diff
@@ -13,7 +13,7 @@ Before((
     'living-status': 'no',
     'tenant-in-uk': 'yes',
-    'current-property-address': 'ABC 222',
+    'tenant-current-address': 'ABC 222',
```
For example - here is now trying to fill out a field that doesn't exist.

lennym commented on apps/right-to-rent-check/index.js:
```diff
@@ -101,9 +101,9 @@ module.exports = {
+    '/tenant-current-address': {
       behaviours: AddressLookup({
-        addressKey: 'current-property-address',
+        addressKey: 'tenant-current-address',
```
This is the change to the field name.

sulthan-ahmed replied:
yeah that was the only one and intentional as mentioned above because 1. I thought it needs to match with the url to get it to work as part of the config. 2. Keep it consistent.

I can change it back if that would help but in most of our apps the step name of any address matches the addressKey

sulthan-ahmed commented:
Ok I get what you're saying. I've now updated the PR name to match what you're referring to.  Looking at my PR, I think I've updated the feature files in accordance with the changes I've made to the addresses

Would you suggest not changing the fields in the future?  I just find it confusing if the address key is different to the step.  I would think it's clearer and helpful when you are searching

lennym commented:
You need to bear in mind that changing field names can have serious side-effects that result in bugs.

Specifically, the customer email is still expecting a field named `current-property-address` that will need to be updated before this can be merged.

sulthan-ahmed commented:
Good point @lennym 

Would you like me to change it back in that case?

lennym commented:
You've come this far... so might as well press on. Just need to update the email code.

sulthan-ahmed commented:
OK, I've learnt my lesson, in the future, I'll avoid changing field names when changing the URLs especially if the service is at a mature state with complex code such as the email service

lennym commented:
The addresses mentioned in the customer emails will also need to be updated:

* https://github.com/UKHomeOffice/right-to-rent-check/blob/master/apps/right-to-rent-check/behaviours/customer-email.js#L17
* https://github.com/UKHomeOffice/right-to-rent-check/blob/master/apps/right-to-rent-check/translations/src/en/email.json#L11-L13

sulthan-ahmed commented:
Nice one @lennym I've updated now

lennym commented:
Will merge once changes from #97 are in.
